yarn-project/circuit-types/src/body.ts

```diff
@@ -1,13 +1,16 @@
 import { L2BlockL2Logs, TxEffect } from '@aztec/circuit-types';
 import { NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP } from '@aztec/circuits.js';
+import { makeTuple } from '@aztec/foundation/array';
+import { padArrayEnd } from '@aztec/foundation/collection';
 import { sha256 } from '@aztec/foundation/crypto';
 import { Fr } from '@aztec/foundation/fields';
-import { BufferReader, serializeToBuffer } from '@aztec/foundation/serialize';
-
-import times from 'lodash.times';
+import { BufferReader, Tuple, serializeToBuffer } from '@aztec/foundation/serialize';
 
 export class Body {
-  constructor(public l1ToL2Messages: Fr[], public txEffects: TxEffect[]) {}
+  constructor(
+    public l1ToL2Messages: Tuple<Fr, typeof NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP>,
```

Comment:
I know I was being annoying with not liking the padding in the archive store but this seem like it will result in the data submitted in calldata to be bigger. But anyway it doesn't matter because withe the new messages model the data will not be here anyway. Just pointing this out here.

Reply:
It will end up being altered again when we get the rest of the message model changes as well 🤷